config/gcppubsub/env_patch.yaml

```diff
@@ -0,0 +1,29 @@
+# This adds the various environment variables needed by the GCP PubSub Source controller. It also
+# adds anything needed by those environment variables (e.g. the secret volume whose location is in
+# a env variable).
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: controller-manager
+spec:
+  template:
+    spec:
+      containers:
+        - name: manager
+          env:
+            - name: ENABLE_GCPPUBSUB_SOURCE
+              value: 'true'
+            - name: GOOGLE_APPLICATION_CREDENTIALS
+              value: /var/secrets/google/key.json
+            - name: GCPPUBSUB_RA_IMAGE
+              value: github.com/knative/eventing-sources/pkg/controller/gcppubsub/receive_adapter
+            - name: GCPPUBSUB_RA_SERVICE_ACCOUNT
+              value: default
```

Comment:
This service account is per-namespace, right?

Reply:
This is added to the K8s Deployment as the Service Account for that Deployment, so it does become namespaced. 

You do raise a good point that users will probably want different service accounts, at least in different namespaces. Maybe the RA's service account should be a field on the source? Or we could make this a map of namespace->serviceAccount. 

Either way, I would like to tackle that in a follow up PR.

Reply:
I lean toward making it a field on the source, but it can be a different PR.

Reply:
Switched to allowing each source being allowed to specify the service account used. If not specified, it should use the normal K8s defaulting mechanism when inserting the Deployment.